File: single_inheritance.py
class Laptop:
    def __init__(self, memory, ssd) -> None:
        self.memory = memory
        self.ssd = ssd

    def coding(self):
        return 'Learning Python and practicing'


class Phone:
    def __init__(self, brand, price, color, origin, dual_sim) -> None:
        self.brand = brand
        self.price = price
        self.color = color
        self.origin = origin
        self.dual_sim = dual_sim

    def phone_call(self, number, text):
        return f'Sending SMS to: {number} with: {text}'

    def __repr__(self) -> str:
        return f'phone: {self.brand} {self.price} {self.dual_sim}'


my_phone = Phone('iphone', 120000, 'silver', 'china', True)
print(my_phone.brand)
print(my_phone)

File: single_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;

class Laptop {
private:
    int memory;
    int ssd;

public:
    Laptop(int memory, int ssd) : memory(memory), ssd(ssd) {}

    string coding() {
        return "Learning Python and practicing";
    }
};

class Phone {
private:
    string brand;
    int price;
    string color;
    string origin;
    bool dual_sim;

public:
    Phone(string brand, int price, string color, string origin, bool dual_sim) : 
        brand(brand), price(price), color(color), origin(origin), dual_sim(dual_sim) {}

    string phone_call(int number, string text) {
        return "Sending SMS to: " + to_string(number) + " with: " + text;
    }

    string toString() {
        return "phone: " + brand + " " + to_string(price) + " " + (dual_sim ? "true" : "false");
    }
};

int main() {
    Phone my_phone("iphone", 120000, "silver", "china", true);
    cout << my_phone.brand << endl;
    cout << my_phone.toString() << endl;
    return 0;
}
